attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java

```diff
@@ -92,17 +97,28 @@ public void destroy(@Observes @Destroyed(ApplicationScoped.class) final Object c
 
     @Override
     public SubscriberBuilder<? extends Message<?>, Void> getSubscriberBuilder(final Config config) {
-        final PubSubConfig pubSubConfig = new PubSubConfig(projectId, getTopic(config), getCredentialPath(config),
+        final PubSubConfig pubSubConfig = new PubSubConfig(getProjectId(config), getTopic(config), getCredentialPath(config),
                 mockPubSubTopics, host.orElse(null), port.orElse(null));
 
         return ReactiveStreams.<Message<?>> builder()
                 .flatMapCompletionStage(message -> CompletableFuture.supplyAsync(() -> {
-                    createTopic(pubSubConfig);
+                    if (isUseAdminClient(config)) {
+                        createTopic(pubSubConfig);
```

Comment:
I would add an INFO message indicating that the topic is going to be created if needed.

Reply:
check `log.adminClientEnabled();`